c7n/resources/appelb.py

```diff
@@ -434,14 +405,13 @@ class AppELBTagAction(tags.Tag):
     batch_size = 1
     permissions = ("elasticloadbalancing:AddTags",)
 
-    def process_resource_set(self, resource_set, ts):
-        _add_appelb_tags(
-            resource_set,
-            self.manager.session_factory,
-            ts)
+    def process_resource_set(self, client, resource_set, ts):
+        client.add_tags(
+            ResourceArns=[alb['LoadBalancerArn'] for alb in resource_set],
```

Comment:
since tagging for target groups is the same as app elbs, we could minimize the diff here even more by pulling the id from the resource and decorating a single tag action for both app elb and target group:

```python
client.add_tags(ResourceArns=[r[self.manager.resource_type.id] for r in resource_set], Tags=ts)
```

and update the docstring accordingly

(same for remove tags)

looks like this applies for app elb, network elb, and target group


>Adds the specified tags to the specified Elastic Load Balancing resource. You can tag your Application Load Balancers, Network Load Balancers, and your target groups.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.add_tags

Reply:
noted and sounds good, but my next steps on this are probably killing/removing tag implementations that have supported by resource group tagging, which will also remove alb/target-group separate implementations.

Reply:
actually given the doc strings  on the subclass seem  useful and the impending changes with universal tag which will also remove, I'm going to leave as is for this pr.